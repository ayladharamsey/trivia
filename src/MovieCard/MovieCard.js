import React from 'react';
import './MovieCard.scss';
import { Link } from 'react-router-dom';

const MovieCard = ({ id, title, episode, release_year, favoriteCards, characters}) => {
  return (
    <section className="card">
        <article className="card-title">
        <h3> Movie Title: {title}</h3>
        </article>
        <article className="card-contents">
        <h3>Episode: <span>{episode}</span></h3>
        <h3>Release Year: <span>{(release_year).slice(0,4)}</span></h3>
        <Link to={`/movies/${id}`} key={id}>
            <button> Characters </button>
        </Link>
        </article>
    </section>
  )
}

export default MovieCard;