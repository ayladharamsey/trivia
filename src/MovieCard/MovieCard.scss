.card_content {
  height: 150px;
}

.card-title {
  height: 50px;
  width: 95%;
  display: flex;
  justify-content: space-between;
}

.star-card-button {
  display: flex;
  height: 50px;
  margin-left: auto;
}

.star-card-button:active {
  display: flex;
  height: 50px;
  margin-left: auto;
}

.residents {
  overflow: scroll;
}