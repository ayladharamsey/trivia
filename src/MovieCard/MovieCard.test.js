import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('MovieCard', () => {
  let wrapper, object ;
  beforeEach(() => {
    
    object = {
      id: 1,
      title: 'starwars',
      episode: 1,
      release_year: "2002",
      characters: ['url.com', 'url.com', 'url.com'],
    }

    wrapper = shallow(<MovieCard 
      id={object.id}
      episode_id={object.episode_id}
      release_year={object.release_year}
      title={object.title}
      characters={object.characters}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});