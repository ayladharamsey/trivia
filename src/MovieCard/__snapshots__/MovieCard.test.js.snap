// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
<section
  className="card"
>
  <article
    className="card-title"
  >
    <h3>
       Movie Title: 
      starwars
    </h3>
  </article>
  <article
    className="card-contents"
  >
    <h3>
      Episode: 
      <span />
    </h3>
    <h3>
      Release Year: 
      <span>
        2002
      </span>
    </h3>
    <Link
      key="1"
      to="/movies/1"
    >
      <button>
         Characters 
      </button>
    </Link>
  </article>
</section>
`;
